fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/FedoraResource.java

```diff
@@ -257,4 +257,10 @@ void replaceProperties(final IdentifierConverter<Resource, FedoraResource> idTra
      * @return the resource being described
      */
     FedoraResource getDescribedResource();
+
+    /**
+     * Get a graph resource corresponding to this resource
+     * @return a graph resource
+     */
+    Resource graphResource(Converter<Resource, String> idTranslator);
```

Comment:
Go for it. Just remember the end goal is an invertible Jena-resource to
internal path string function that doesn't load nodes in the common
execution path, passed down to the property resolution bits.

On Aug 4, 2016 7:23 PM, "A. Soroka" <notifications@github.com> wrote:

> In fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/
> models/FedoraResource.java
> <https://github.com/fcrepo4/fcrepo4/pull/1086#discussion_r73620144>:
>
> > @@ -257,4 +257,10 @@ void replaceProperties(final IdentifierConverter<Resource, FedoraResource> idTra
> >       * @return the resource being described
> >       */
> >      FedoraResource getDescribedResource();
> > +
> > +    /**
> > +     * Get a graph resource corresponding to this resource
> > +     * @return a graph resource
> > +     */
> > +    Resource graphResource(Converter<Resource, String> idTranslator);
>
> A lot of what I was talking about is in there, but some is not and some is
> not addressed in the way I meant (not your fault, my fault for using
> unclear language). What do you say to me sending you a PR to this PR with
> my highly opinionated opinions? Then you can cherrypick from that what you
> like.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fcrepo4/fcrepo4/pull/1086/files/f4c90a8742a4337d6df3f673276e13abd4da98c4#r73620144>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHUNHxDdJEYpylpKFnTUtt2FhwTHta1ks5qcnRigaJpZM4JdBw8>
> .
>